services/net/transcription/TranscriptionManager.cs

```diff
@@ -232,40 +232,50 @@ private async Task UpdateTranscriptionAsync(ContentModel content)
         // Remote storage locations may not be easily accessible by this service.
         var path = content.FileReferences.FirstOrDefault()?.Path;
         var safePath = Path.Join(_options.VolumePath, path.MakeRelativePath());
+       
         if (File.Exists(safePath))
         {
             this.Logger.LogInformation("Transcription requested.  Content ID: {Id}", content.Id);
 
-            var original = content.Body;
-            var fileBytes = File.ReadAllBytes(safePath);
-
-            var transcript = await RequestTranscriptionAsync(fileBytes); // TODO: Extract language from data source.
+            // convert to audio if it's video file
+            var isVideo = Path.GetExtension(safePath).ToLower() == ".mp4";
+            if (isVideo) {
+                var convertFilePath = await Video2Audio(safePath);
+                safePath = convertFilePath;
+            }
 
-            // Fetch content again because it may have been updated by an external source.
-            // This can introduce issues if the transcript has been edited as now it will overwrite what was changed.
-            var result = await _api.FindContentByIdAsync(content.Id);
-            if (result != null && !String.IsNullOrWhiteSpace(transcript))
+            if (!isVideo || !String.IsNullOrEmpty(safePath))
```

Comment:
If `Video2Audio` returns an empty string then we wouldn't perform this process.  Simplify to `!String.IsNullOrEmpty(safePath)`.